src/components/docs/date-time-picker.js

```diff
@@ -0,0 +1,406 @@
+    },
+  },
+  data: {
+    date: new Date().toISOString().substr(0, 10),
+  },
+};
+
+export const DateUiSchemaOptions = () => (
+  <Demo
+    data={DateSchemaUiOptionsInput.data}
+    schema={DateSchemaUiOptionsInput.schema}
+    uischema={DateSchemaUiOptionsInput.uischema}
+  />
+);
+
+export const DateTimeSchemaInput = {
+  schema: {
+    properties: {
+      datetime: {
+        type: 'string',
+        format: 'date-time',
+        description: 'schema-based datetime picker'
+      }
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/datetime',
+  },
+  data: {
+    datetime: new Date().toISOString()
+  },
+};
+
+export const DateTimeSchema = () => (
+  <Demo
+    data={DateTimeSchemaInput.data}
+    schema={DateTimeSchemaInput.schema}
+    uischema={DateTimeSchemaInput.uischema}
+  />
+);
+
+export const DateTimeUiSchemaInput = {
+  schema: {
+    properties: {
+      datetime: {
+        type: 'string',
+        description: 'uischema-based datetime picker'
+      }
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/datetime',
+    options: {
+      format: 'date-time'
+    }
+  },
+  data: {
+    datetime: new Date().toISOString()
+  },
+};
+
+export const DateTimeUiSchema = () => (
+  <Demo
+    data={DateTimeUiSchemaInput.data}
+    schema={DateTimeUiSchemaInput.schema}
+    uischema={DateTimeUiSchemaInput.uischema}
+  />
+);
+
+export const DateTimeUiOptionSchemaInput = {
+  schema: {
+    properties: {
+      datetime: {
+        type: 'string',
+        description: 'uischema-based datetime picker'
+      }
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/datetime',
+    options: {
+      format: 'date-time',
+      clearLabel: 'Clear it!',
+      cancelLabel: 'Abort',
+      okLabel: 'Do it',
+      dateTimeFormat: 'DD-MM-YY hh:mm:a',
+      dateTimeSaveFormat: 'YYYY/MM/DD h:mm a',
+      ampm: true
+    }
+  },
+  data: {
+    datetime: new Date().toISOString()
+  },
+};
+
+export const DateTimeUiOptionSchema = () => (
+  <Demo
+    data={DateTimeUiOptionSchemaInput.data}
+    schema={DateTimeUiOptionSchemaInput.schema}
+    uischema={DateTimeUiOptionSchemaInput.uischema}
+  />
+);
+
+export const TimeOptionTable = () => (
+  <div>
+    <h2>Time Picker Options</h2>
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableCell>Option</TableCell>
+          <TableCell>Description</TableCell>
+        </TableRow>
+      </TableHead>
+      <TableBody>
+        <TableRow>
+          <TableCell>ampm</TableCell>
+          <TableCell>If set to true, 12-hour format is used, otherwise 24-hour time format is used.</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>timeFormat</TableCell>
+          <TableCell>Defines the used format</TableCell>
+        </TableRow>
+        <TableRow>
+          <TableCell>timeSaveFormat</TableCell>
+          <TableCell>Defines the save format of the time</TableCell>
```

Comment:
We should explain some of the pitfalls of the `timeSaveFormat` option.

```suggestion
          <TableCell>The format in which the time is saved in the data. Note that if you specify a format which is incompatible with JSON Schema's "time" format then
           you should use the UI Schema based invocation, otherwise the control will be marked with an error.</TableCell>
```